pkg/util/walkers/copying.go

```diff
@@ -36,16 +37,24 @@ func CopyingWalker(path string, newDirPerms os.FileMode) assets.WalkFunc {
 			return errors.Wrap(err, "failed to create dir")
 		}
 
-		// TODO: If we start packing binaries, make sure they have executable bit set.
-		targetFile, err := os.OpenFile(fileName, os.O_RDWR|os.O_CREATE, 0644)
-		if err != nil {
-			return errors.Wrap(err, "failed to open target file")
-		}
-		defer targetFile.Close()
+		return writeFile(fileName, r)
+	}
+}
 
-		if _, err := io.Copy(targetFile, r); err != nil {
-			return errors.Wrap(err, "failed to write file")
-		}
-		return nil
+// writeFile writes data from given io.Reader to the file and makes sure, that
+// this is the only content stored in the file.
+func writeFile(p string, r io.Reader) error {
+	// TODO: If we start packing binaries, make sure they have executable bit set.
+	f, err := os.OpenFile(p, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to open target file %s: %v", p, err)
 	}
+
```

Comment:
I like having the defer right after the previous statement without any empty lines, is this something enforced by the linter?

Reply:
Fixed.